src/Agent.Plugins/TestFilePublisher/Plugin/TestFilePublisherLogPlugin.cs

```diff
@@ -0,0 +1,208 @@
+            {
+                return false;
+            }
+
+            // Enable only for build
+            if (!context.Variables.TryGetValue("system.hosttype", out var hostType)
+                || !string.Equals("Build", hostType.Value, StringComparison.OrdinalIgnoreCase))
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "NotABuild");
+                return true;
+            }
+
+            // Disable for on-prem
+            if (!context.Variables.TryGetValue("system.servertype", out var serverType)
+                || !string.Equals("Hosted", serverType.Value, StringComparison.OrdinalIgnoreCase))
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "NotHosted");
+                return true;
+            }
+
+            // check for PTR task or some other tasks to enable/disable
+            if (context.Steps == null)
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "NoSteps");
+                return true;
+            }
+
+            if (context.Steps.Any(x => x.Id.Equals(new Guid("0B0F01ED-7DDE-43FF-9CBB-E48954DAF9B1"))))
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "ExplicitPublishTaskPresent");
+                return true;
+            }
+
+            if (_pipelineConfig.BuildId == 0)
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "BuildIdZero");
+                return true;
+            }
+
+            if (_pipelineConfig.Pattern == null)
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "PatternIsEmpty");
+                return true;
+            }
+
+            if (_pipelineConfig.SearchFolders == null)
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "SearchFolderIsEmpty");
+                return true;
+            }
+
+            return false;
+        }
+
+        private async Task PopulatePipelineConfig(IAgentLogPluginContext context)
+        {
+            var props = new Dictionary<string, Object>();
+
+            if (context.Variables.TryGetValue("system.teamProject", out var projectName))
+            {
+                _pipelineConfig.ProjectName = projectName.Value;
+                _telemetry.AddOrUpdate("ProjectName", _pipelineConfig.ProjectName);
+                props.Add("ProjectName", _pipelineConfig.ProjectName);
+            }
+
+            if (context.Variables.TryGetValue("build.buildId", out var buildId))
+            {
+                _pipelineConfig.BuildId = int.Parse(buildId.Value);
+                _telemetry.AddOrUpdate("BuildId", _pipelineConfig.BuildId);
+                props.Add("BuildId", _pipelineConfig.BuildId);
+            }
+
+            if (context.Variables.TryGetValue("System.DefinitionId", out var buildDefinitionId))
+            {
+                _telemetry.AddOrUpdate("BuildDefinitionId", buildDefinitionId.Value);
```

Comment:
can you check if this is working, i added this for trp but not showing up in telemetry

Reply:
still pending right?

---
In reply to: [263670330](https://github.com/Microsoft/azure-pipelines-agent/pull/2141#discussion_r263670330) [](ancestors = 263670330)